hibp.js

```diff
@@ -1,50 +1,81 @@
 "use strict";
 
 const got = require("got");
+const createDOMPurify = require("dompurify");
+const { JSDOM } = require("jsdom");
 
 const AppConstants = require("./app-constants");
-const DBUtils = require("./db/utils");
 const pkg = require("./package.json");
 
 
+const DOMPurify = createDOMPurify((new JSDOM("")).window);
 const HIBP_USER_AGENT = `${pkg.name}/${pkg.version}`;
 
 
 const HIBP = {
-  async req(path, options = {}) {
-    // Construct HIBP url and standard headers
-    const url = `${AppConstants.HIBP_KANON_API_ROOT}${path}?code=${encodeURIComponent(AppConstants.HIBP_KANON_API_TOKEN)}`;
+  _addStandardOptions (options = {}) {
     const hibpOptions = {
       headers: {
         "User-Agent": HIBP_USER_AGENT,
       },
       json: true,
     };
-    const reqOptions = Object.assign(options, hibpOptions);
+    return Object.assign(options, hibpOptions);
+  },
+
+  async req(path, options = {}) {
+    const url = `${AppConstants.HIBP_API_ROOT}${path}?code=${encodeURIComponent(AppConstants.HIBP_API_TOKEN)}`;
+    const reqOptions = this._addStandardOptions(options);
     return await got(url, reqOptions);
   },
 
-  async getBreachesForEmail(sha1) {
+  async kAnonReq(path, options = {}) {
+    // Construct HIBP url and standard headers
+    const url = `${AppConstants.HIBP_KANON_API_ROOT}${path}?code=${encodeURIComponent(AppConstants.HIBP_KANON_API_TOKEN)}`;
+    const reqOptions = this._addStandardOptions(options);
+    return await got(url, reqOptions);
+  },
+
+  async loadBreachesIntoApp(app) {
+    console.log("Loading breaches from HIBP into app.locals");
+    try {
+      const breachesResponse = await this.req("/breaches");
+      const breaches = [];
+
+      for (const breachIndex in breachesResponse.body) {
```

Comment:
Should this be `for (const breach of breachesResponse.body) {...}`?
Then we shouldn't need the intermediate `breachIndex` and `breachesResponse.body[breachIndex]` stuffs.

Reply:
Weird. I thought I started with that but the `got` return value wasn't iterable. I just tried it with `of` and it seems to work fine now. 🤷‍♂️